common/wireaddr.c

```diff
@@ -299,13 +303,19 @@ bool wireaddr_from_hostname(struct wireaddr *addr, const char *hostname,
 		return true;
 	}
 
+	/* Tell them we wanted DNS and fail. */
+	if (no_dns) {
+		if (err_msg)
+			*err_msg = "Needed DNS, but lookups suppressed";
+		*no_dns = true;
+		return false;
+	}
+
 	memset(&hints, 0, sizeof(hints));
```

Comment:
Thanks! for doing this, confidence in that we care will give us later the feedback ln needs